File: app/src/main/java/quintype/com/templatecollectionwithrx/models/story/ShrubberyUserEngagement.java

package quintype.com.templatecollectionwithrx.models.story;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class ShrubberyUserEngagement {

    @SerializedName("views")
    @Expose
    private Integer views;

    public Integer getViews() {
        return views;
    }

    public void setViews(Integer views) {
        this.views = views;
    }

}

File: app/src/main/java/quintype/com/templatecollectionwithrx/models/story/ShrubberyUserEngagement.kt
package quintype.com.templatecollectionwithrx.models.story

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Created TemplateCollectionWithRx by rakshith on 9/4/18.
 */

class ShrubberyUserEngagement() : Parcelable {
    override fun describeContents(): Int {
        return 0
    }

    @SerializedName("views")
    @Expose
    var views: Int? = null

    constructor(parcel: Parcel) : this() {
        views = parcel.readValue(Int::class.java.classLoader) as? Int
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeValue(views)
    }

    companion object CREATOR : Parcelable.Creator<ShrubberyUserEngagement> {
        override fun createFromParcel(parcel: Parcel): ShrubberyUserEngagement {
            return ShrubberyUserEngagement(parcel)
        }

        override fun newArray(size: Int): Array<ShrubberyUserEngagement?> {
            return arrayOfNulls(size)
        }
    }

}